### ISSUE TITLE
command line options for VTK output?

### ISSUE THREAD

**Issue by kingaj12** [OPM/opm-simulators#1421]
Hi - I am trying to write vtk output, but can't find any documentation on how to do this from the command line.  I can probably figure out how to hack the code to do it, but is there an easier way?

Also - another topic - it seems like "flow --help" should be supported, but it throws an error.

**Comment by GitPaean** [comment]
Hi, under the output folder, you will find a hidden file filename begin with `.`.  For example, if you deck file is SPE1CASE1.DATA, you will find a file `.SPE1CASE1.DEBUG` under the output folder. 

In the beginning of that file, you will find all the running options, like `output_dir`, `max_iter`. You can use them like something 

`flow SPE1CASE1.DATA output_dir=abc max_iter=200` to set the running options. 

However, I did not find anything related to `VTK` output either. Maybe it is something more involved in `ewoms`? is this true, @andlaus ?

There is another software available for you to visualize the simulation result. 
http://resinsight.org/

It is specialized for visualizing reservoir simulation, many people use it  to analyze flow and eclipse simulation results, maybe you will find it is useful.

**Comment by andlaus** [comment]
> However, I did not find anything related to VTK output either. Maybe it is something more involved in ewoms? is this true, @andlaus ?

yes, that is the case: the default format for simulation output in eWoms is VTK, and I use it as all the time when I develop features for `ebos` because it is less hassle to extend than the ECL format. To get VTK output with `ebos`, you need to pass the `--enable-vtk-output=true` command line parameter. Since there is no way to set this parameter from `flow`, some glue code analogous to the `--ecl-output-dir=` stuff needs to be added to `flow` to make it work.

**Comment by atgeirr** [comment]
It turns out VTK output capability (used to be triggered by output_vtk=true on the command line or in a parameter file) was removed from flow in PR #1377. I did not realize that implication at the time, which is unfortunate, as I also need VTK for another project!

Not sure what is the best way to restore it. Using the ewoms duplicate functionality would be one way.

**Comment by kingaj12** [comment]
It looks like command line arguments are processed in two places.  Once in flow.cpp [https://github.com/OPM/opm-simulators/blob/4a6ab7ac76dafe1014b55ad95f714a6abd6d1bc8/examples/flow.cpp#L98](https://github.com/OPM/opm-simulators/blob/4a6ab7ac76dafe1014b55ad95f714a6abd6d1bc8/examples/flow.cpp#L98)
and then (in my application) in the execute() method of flow_ebos_blackoil.cpp:
[https://github.com/OPM/opm-simulators/blob/4a6ab7ac76dafe1014b55ad95f714a6abd6d1bc8/opm/simulators/flow_ebos_blackoil.cpp#L59](https://github.com/OPM/opm-simulators/blob/4a6ab7ac76dafe1014b55ad95f714a6abd6d1bc8/opm/simulators/flow_ebos_blackoil.cpp#L59)

It seems like parameters passed into flow are then ignored by the subsequent call to flow_ebos_xxx.  Perhaps this is where to look?

**Comment by andlaus** [comment]
> Using the ewoms duplicate functionality would be one way.

it would be by far the easiest way; all you need to do is to do the same thing as for `output_dir` to `output_vtk`:

https://github.com/OPM/opm-simulators/blob/master/opm/autodiff/FlowMainEbos.hpp#L402

note that the ewoms VTK code allows to enable and disable individual data fields.  while this can be done the same way as for the other parameters, `flow_legacy` never supported anything compareable, so we should probably not bother.
